src/vmod_xkey.c

```diff
@@ -602,14 +608,21 @@ vmod_event(VRT_CTX, struct vmod_priv *priv, enum vcl_event_e e)
 	switch (e) {
 	case VCL_EVENT_LOAD:
 		AZ(pthread_mutex_lock(&mtx));
-		if (n_init == 0)
+		if (n_init == 0) {
 #ifdef HAVE_ENUM_EXP_EVENT_E
 			xkey_cb_handle =
 			    EXP_Register_Callback(xkey_cb, NULL);
 #else
 			xkey_cb_handle = ObjSubscribeEvents(xkey_cb, NULL,
 			    OEV_INSERT|OEV_EXPIRE);
 #endif
+			if (vsc == NULL) {
```

Comment:
Do you mean that we should assert that vsc is NULL, like this?
```suggestion
			AZ(vsc);
```

Reply:
No, that's not right. vsc will be non-NULL if we're loading a new VCL and both the new and old VCL use vmod_xkey.

Reply:
I think the change is right, this shouldn't happen twice being guarded by `n_init`, and it should really be an invariant so an assert is more relevant than a branch.

Maybe it would be better with test case coverage, we'd be... certain ;)

Reply:
> this shouldn't happen twice being guarded by `n_init`

Ah, I see what you mean. `if (vsc == NULL) {` is redundant when it's already inside `if (n_init == 0) {`.

I'll make that change and write a test.